app/models/analyzer.rb

```diff
@@ -0,0 +1,85 @@
+class Analyzer
+  INCREMENT_NUM_PER_TEXT = 100
+
+  def initialize(use_persistent: false)
+    @uri = URI.parse("#{Rails.configuration.elasticsearch[:host]}/entries/_analyze")
+    @http =
+      if use_persistent
+        Net::HTTP::Persistent.new
+      else
+        Net::HTTP.new(@uri.host, @uri.port)
+      end
+    @post = Net::HTTP::Post.new(@uri.request_uri, 'Content-Type' => 'application/json')
+    @use_persistent = use_persistent
+  end
+
+  def normalize(text, normalizer)
+    raise ArgumentError, "Empty text" unless text.present?
+    _text = text.tr('{}', '()')
+    body = { analyzer: normalizer, text: _text }.to_json
+    response = tokenize(body)
+
+    (JSON.parse response.body, symbolize_names: true)[:tokens].map{ _1[:token] }.join('')
+  end
+
+  def batch_normalize(texts, normalizer)
+    ## Explanation
+    # This method returns the following results from input texts corresponding to normalizer.
+    # texts:              ["abc def", "of", "ghi"]
+    # normalize1 results: ["abcdef", "of", "ghi"]
+    # normalize2 results: ["abcdef", "", "ghi"]
+
+    raise ArgumentError, "Empty text in array" unless texts.present?
```

Comment:
今更、気がつきましたが、texts配列の中をみる必要がありそうです。
```suggestion
    raise ArgumentError, "Empty text in array" if texts.any?{ _1.empty? }
```
のようなイメージです。

Reply:
配列の中身に加えて配列自体もチェックするように修正しました。
また、#batch_normalizeが`if`に変わったので#normalizeの方も合わせて記述方法を変更しました。

https://github.com/pubannotation/pubdictionaries/pull/157/commits/ebb6b946a2ff101cf7bf82075289787fcac794ff